src/util/websocket/WebSocketManager.cpp

```diff
@@ -0,0 +1,241 @@
+//  Copyright 2023, University of Freiburg,
+//  Chair of Algorithms and Data Structures.
+//  Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#include "WebSocketManager.h"
+
+#include <absl/cleanup/cleanup.h>
+#include <absl/container/btree_map.h>
+#include <absl/container/flat_hash_map.h>
+#include <ctre/ctre.h>
+
+#include <boost/asio/experimental/awaitable_operators.hpp>
+#include <iostream>
+#include <mutex>
+#include <optional>
+
+#include "../MultiMap.h"
+#include "../http/HttpUtils.h"
+#include "./Common.h"
+#include "./QueryState.h"
+
+namespace ad_utility::websocket {
+
+class WebSocketId {
+  uint32_t id_;
+  explicit WebSocketId(uint32_t id) : id_{id} {}
+
+ public:
+  // Generate unique webSocketIds until it overflows and hopefully the old
+  // ids are gone by then.
+  static WebSocketId uniqueId() {
+    static std::atomic<uint32_t> webSocketIdCounter{0};
```

Comment:
As discussed we might want to consider an address based id assignment system

Reply:
We definitely want to do this. 
Just make sure, that the address is stable. But do you need the address at all, or can you just pass the callbacks directly? I will see.